### PR TITLE
Drop support for passing a string to the `title` attribute, and drop support for deprecated attributes `titlefont`, `titleposition`, `titleside`,  and `titleoffset`

### PR DESCRIPTION
**Pull request by emilykl**
Closes: #7177 

- Drop support for passing a string to the `title` attribute (use `title.text` instead)
- Drop support for deprecated `titlefont` attribute (use `title.font` instead)
- Drop support for deprecated `titleposition` attribute (use `title.position` instead)
- Drop support for deprecated `titleside` attribute (use `title.side` instead)
- Drop support for deprecated `titleoffset` attribute (use `title.offset` instead)
- Apply these changes apply to everywhere the above attributes appear(`layout`, `colorbar`, `axis`, etc.)

Also:
- Remove jasmine tests corresponding to the removed attributes
- Remove functions used for converting old attributes to new attributes

### CODE REVIEW

File: src/plots/gl3d/layout/axis_attributes.js [hunk 1]
```diff
@@ -122,7 +122,5 @@ module.exports = overrideAll({
     zerolinecolor: axesAttrs.zerolinecolor,
     zerolinewidth: axesAttrs.zerolinewidth,
     _deprecated: {
```
Comment (archmoj) [inline src/plots/gl3d/layout/axis_attributes.js:124]: Please remove the empty `_deprecated` key as well.
Thank you!

File: src/plots/layout_attributes.js [hunk 2]
```diff
@@ -461,20 +452,5 @@ module.exports = {
         editType: 'none'
     }),
     _deprecated: {
```
Comment (archmoj) [inline src/plots/layout_attributes.js:454]: Also here please remove the empty `_deprecated` key.

File: src/plots/ternary/layout_attributes.js [hunk 3]
```diff
@@ -63,8 +63,6 @@ var ternaryAxesAttrs = {
         ].join(' ')
     },
     _deprecated: {
```
Comment (archmoj) [inline src/plots/ternary/layout_attributes.js:65]: :hocho:  _deprecated

File: src/traces/pie/attributes.js [hunk 4]
```diff
@@ -303,28 +294,5 @@ module.exports = {
     },
 
     _deprecated: {
```
Comment (archmoj) [inline src/traces/pie/attributes.js:296]: :hocho:  _deprecated

**Comment by emilykl** [comment]
@archmoj Thanks very much for the review. I've removed the empty `_deprecated` keys. Do you have any other comments?

**Review by archmoj — APPROVED** [review]
:dancer:

**Comment by archmoj** [comment]
@emilykl Do you need my help resolving the conflicts with master?

**Comment by emilykl** [comment]
> @emilykl Do you need my help resolving the conflicts with master?

@archmoj Nope, all good!

**Comment by archmoj** [comment]
:dancer:

**Pull request merged**